### ISSUE TITLE
Duplicate friend list

### ISSUE THREAD

**Issue by mrpond** [rathena/rathena#8311]
### rAthena Hash

a44d9540fb28cad1ac3da07f512fece15cbf00d8

### Client Date

20131120

### Server Mode

Pre-Renewal, Renewal

### Result

character server will print error when try to insert duplicate friend id into table

### Relevant Log Output

```Shell
N/A
```

### Expected Result

fix
rAthena\src\char\char.cpp
```
int char_mmo_char_tosql(uint32 char_id, struct mmo_charstatus* p){
...
		StringBuf_Printf(&buf, "INSERT INTO `%s` (`char_id`, `friend_id`) VALUES ", schema_config.friend_db);
//===
		for (size_t index_friend = 0; index_friend < MAX_FRIENDS; ++index_friend) {
			if (p->friends[index_friend].char_id > 0) {
				for (size_t lookup = 0; lookup < MAX_FRIENDS; ++lookup) {
					if (index_friend != lookup) {
						if (p->friends[index_friend].char_id == p->friends[lookup].char_id) {
							p->friends[lookup].char_id = 0;
						}
					}
				}
			}
		}
//===
		for( i = 0, count = 0; i < MAX_FRIENDS; ++i )
...
```

### How to Reproduce

N/A

### Official Information

N/A

**Comment by Lemongrass3110** [comment]
Originally reported in #5277. We did not get feedback to our questions there.

Since we do not know how to reproduce it, we are unable to fix this.

Do you know the necessary steps to reproduce this issue?

**Comment by mrpond** [comment]
> Originally reported in #5277. We did not get feedback to our questions there.
> 
> Since we do not know how to reproduce it, we are unable to fix this.
> 
> Do you know the necessary steps to reproduce this issue?

no, I don't know how to reproduce it. error happen when character to save the data into table.
Server had around 1800 players. and it happen randomly
